sympy/sets/fancysets.py

```diff
@@ -577,190 +600,146 @@ def reversed(self):
         if not self:
             return self
         return self.func(
-            self.stop - self.step, self.start - self.step, -self.step)
+            self._stop - self.step, self.start - self.step, -self.step)
 
     def _contains(self, other):
+        from sympy.functions.elementary.piecewise import Piecewise
+        other = _sympify(other)
         if not self:
             return S.false
         if other.is_infinite:
             return S.false
         if not other.is_integer:
             return other.is_integer
-        ref = self.start if self.start.is_finite else self.stop
-        if (ref - other) % self.step:  # off sequence
-            return S.false
-        return _sympify(other >= self.inf and other <= self.sup)
+        ref = self.start if self.start.is_finite else self._stop
+        return Piecewise((False, Ne((ref - other) % self.step, S.Zero)),
+                         (True, Ge(other, self._inf) & Le(other, self._sup)),
+                         (False, True))
 
     def __iter__(self):
+        # print([param.is_Symbol for param in (self.start, self._stop, self.step)])
+        # if any(param.is_Symbol for param in (self.start, self._stop, self.step)):
+        #     raise ValueError("Cannot iterate over symbolic Range.")
         if self.start in [S.NegativeInfinity, S.Infinity]:
             raise ValueError("Cannot iterate over Range with infinite start")
-        elif self:
+        if self.size.is_Integer == False and self.size not in [S.NegativeInfinity, S.Infinity]:
```

Comment:
The query `x.is_Integer` should be the same as `isinstance(x, Integer)` so it will always give True or False. Queries like x.is_integer on the other hand use ternary logic and can give None. In ternary logic it is reasonable to write x.is_integer == False to distinguish between False and None but in the binary case that isn't needed so this can just be
```
if not self.size.is_Integer and self.size not in [S.NegativeInfinity, S.Infinity]
```
Then I'd rather factor that to
```
if not (self.size.is_Integer or self.size in [S.NegativeInfinity, S.Infinity]):
```
but that's not important